File: Easy/Java/1941. Check if All Characters Have Equal Number of Occurrences.java
class Solution {
    public boolean areOccurrencesEqual(String s) {
        int[] count = new int[26];
        int max = 0;
        
        for(int i = 0; i < s.length(); i++){
            count[s.charAt(i) - 'a']++;
        }

        for(int x : count){
            max = Math.max(x, max);
        }

        for(int i = 0; i < count.length; i++){
            if(count[i] != 0 && max != count[i])
                return false;
        }

        return true;
    }
}

File: Easy/JavaScript/1941. Check if All Characters Have Equal Number of Occurrences.js
var areOccurrencesEqual = function(s) {
    let freq = {};
    
    for(let c of s){
        freq[c] = (freq[c] || 0) + 1;
    }

    let max = freq[s[0]];
    for(let c in freq){
        if(freq[c] != max)
            return false;
    }

    return true;
};
